Source/WebCore/svg/animation/SVGSMILElement.cpp

```diff
@@ -1179,6 +1182,14 @@ bool SVGSMILElement::progress(SMILTime elapsed, SVGSMILElement& firstAnimation,
     if (seekToTime) {
         if (m_activeState == Inactive || m_activeState == Frozen)
             smilEventSender().dispatchEventSoon(*this, eventNames().endEventEvent);
+
+        if (repeat) {
+            for (unsigned repeatEventCount = 1; repeatEventCount < repeat; repeatEventCount++)
```

Comment:
The local variable name here is strange. It’s named with the word "count" but it’s not a count; it’s a loop index. The value `repeat` is actually the count. I think it would be better otherwise just use a traditional `i` or something like that for the loop index.

The loop here is strangely written. If `repeat` is 3, this will emit dispatch 2 repeat events, for values 1 and 2. Starting with 1 means this loop dispatches `repeat - 1` events. If this is intentional, then we need a comment explaining it; perhaps it’s supposed to dispatch `repeat - 1` events, and the dispatch outside the loop dispatches the last one? Either way, we also need a test that involves multiple repeat events, proving that it’s correct. I’m not sure we have one.

A more traditional way to write the loop would be:
```
for (unsigned i = 0; I < repeat - 1; ++i)
```
But we’d still need a comment explaining why we are doing one less than `repeat`.

Reply:
Yes - repeat(n) would be next step to fix. I just merged first part and then next would be repeat(n) case. :-)

We have repeatn cases in WPT and we fail them currently. :-(

Reply:
It also seems like we potentially repeat in the `if (animationIsContributing) {` code path (prior to this), so we might have repeated one time already. That might be why this does one fewer than the repeat count, but that would only be true if `animationIsContributing` is true, and that's not checked here.

Then (as Darin points out) we potentially repeat again if `m_activeState == Inactive`, so it seems like all of this code would tend to repeat more times than expected.